aws/rust-runtime/aws-config/src/default_provider.rs

```diff
@@ -89,17 +86,74 @@ pub mod region {
     }
 }
 
+/// Default retry behavior configuration provider chain
+pub mod retry_config {
+    use smithy_types::retry::RetryConfig;
+
+    use crate::environment::retry_config::EnvironmentVariableRetryConfigProvider;
+    use crate::profile;
+    use crate::provider_config::ProviderConfig;
+
+    /// Default RetryConfig Provider chain
+    ///
+    /// This provider will check the following sources in order:
+    /// 1. [Environment variables](EnvironmentVariableRetryConfigProvider)
+    /// 2. [Profile file](crate::profile::retry_config::ProfileFileRetryConfigProvider)
+    pub fn default_provider() -> Builder {
```

Comment:
code example would probably be helpful here since the name of the returned type is a little surprising